driver-core/src/test/functional/com/mongodb/internal/connection/DefaultConnectionPoolTest.java

```diff
@@ -399,7 +401,12 @@ public void checkoutHandOverMechanism() throws InterruptedException, TimeoutExce
         provider.ready();
         List<InternalConnection> connections = new ArrayList<>();
         for (int i = 0; i < openConnectionsCount; i++) {
-            connections.add(provider.get(OPERATION_CONTEXT, 0, NANOSECONDS));
+            OperationContext operationContextWithZeroMaxWait = new OperationContext(
+                    OPERATION_CONTEXT.getRequestContext(),
+                    OPERATION_CONTEXT.getSessionContext(),
+                    new TimeoutContext(new TimeoutSettings(30_000, 10_000, 0, null, 0)),
+                    OPERATION_CONTEXT.getServerApi());
+            connections.add(provider.get(operationContextWithZeroMaxWait));
```

Comment:
Note that this constructs a new operation context, as before with a 0 wait time (which I believe is intentional in this test), rather than just using OPERATION_CONTEXT.

Reply:
Ack